website/pages/en/network/developing.mdx

```diff
@@ -2,52 +2,104 @@
 
-Once a subgraph has been processed by Indexers and is available for querying, developers can start to use the subgraph in their applications. Developers query subgraphs via a gateway, which forwards their queries to an Indexer who has processed the subgraph, paying query fees in GRT.
+- `subgraph.yaml`: this YAML file contains the [subgraph manifest](/developing/creating-a-subgraph/#the-subgraph-manifest)
+- `subgraph.graphql`: this GraphQL schema defines what data is stored for your subgraph, and how to query it via [GraphQL](/developing/creating-a-subgraph/#the-graphql-schema)
+- `AssemblyScript Mappings`: [AssemblyScript](https://github.com/AssemblyScript/assemblyscript) mappings translates data from the event data to the entities defined in your schema
 
-In order to make queries, developers must generate an API key, which can be done in Subgraph Studio. This API key must be funded with GRT, in order to pay query fees. Developers can set a maximum query fee, in order to control their costs, and limit their API key to a given subgraph or origin domain. Subgraph Studio provides developers with data on their API key usage over time.
+Learn the detailed specifics to [create a subgraph](/developing/creating-a-subgraph/).
 
-Developers are also able to express an Indexer preference to the gateway, for example preferring Indexers whose query response is faster, or whose data is most up to date. These controls are set in Subgraph Studio.
+### Subgraph Lifecycle
 
-### Updating Subgraphs
+Here is a general view of a subgraph’s cycle:
 
-After a time a subgraph developer may want to update their subgraph, perhaps fixing a bug or adding new functionality. The subgraph developer may deploy new version(s) of their subgraph to Subgraph Studio for rate-limited development and testing.
+![Subgraph Lifecycle](/img/subgraph-lifecycle.png)
 
-Once the Subgraph Developer is ready to update, they can initiate a transaction to point their subgraph at the new version. Updating the subgraph migrates any signal to the new version (assuming the user who applied the signal selected "auto-migrate"), which also incurs a migration tax. This signal migration should prompt Indexers to start indexing the new version of the subgraph, so it should soon become available for querying.
+### Context for Subgraph Development
 
-### Deprecating Subgraphs
+Great subgraphs start with a local development environment and unit tests. Developers can use `graph-cli,` `graph-ts,` and [Matchstick](/developing/unit-testing-framework/) to build robust subgraphs that deploy to The Graph Network.
 
-At some point a developer may decide that they no longer need a published subgraph. At that point they may deprecate the subgraph, which returns any signalled GRT to the Curators.
+> Developers use [Graph CLI](https://github.com/graphprotocol/graph-tooling/tree/main/packages/cli), a command line interface tool for building and deploying subgraphs in The Graph.
```

Comment:
This was mentioned in the previous paragraph (spelled `graph-cli`; it's the same thing), so I would maybe merge those paragraphs?